### PR TITLE
⭐️ atlassian provider

### PR DESCRIPTION
**Pull request by mariuskimmina**
Reopening #2042 from a branch instead of a fork

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -313,6 +314,9 @@ providers/build/aws: providers/lr
 providers/install/aws:
 	@$(call installProvider, providers/aws)
 
+providers/build/atlassian: providers/lr
```
Comment (imilchev) [inline Makefile:318]: can you add the `providers/install/atlassian` target too

File: providers/atlassian/connection/admin/connection.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+type AdminConnection struct {
+	id     uint32
+	Conf   *inventory.Config
+	asset  *inventory.Asset
+	client *admin.Client
+	host   string
+}
+
+func NewConnection(id uint32, asset *inventory.Asset, conf *inventory.Config) (*AdminConnection, error) {
+	adminToken := conf.Options["admintoken"]
```
Comment (imilchev) [inline providers/atlassian/connection/admin/connection.go]: this won't be set. Same applies to the other options you assume in the other connection types. You should define the flags that are needed for these and handle them

**Review by imilchev — COMMENTED** [review]
Looks good! Almost there. Added a few small things, also need to check the failing Github actions 🙌

File: providers/atlassian/connection/admin/connection.go (hunk 2 above)
Comment (mariuskimmina) [reply providers/atlassian/connection/admin/connection.go]: They should be working now

File: providers/atlassian/connection/admin/connection.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+	if adminToken == "" {
+		adminToken = os.Getenv("ATLASSIAN_ADMIN_TOKEN")
+	}
+	if adminToken == "" {
+		return nil, errors.New("you need to provide atlassian admin token via ATLASSIAN_ADMIN_TOKEN env")
```
Comment (imilchev) [inline providers/atlassian/connection/admin/connection.go]: this error is no longer correct. Either the flag or the var needs to be set. Make sure that's updated for the other flags too

File: providers/atlassian/config/config.go [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+	Connectors: []plugin.Connector{
+		{
+			Name:      "atlassian",
+			Use:       "atlassian",
+			Short:     "atlassian",
+			MaxArgs:   2,
+			Discovery: []string{},
+			Flags: []plugin.Flag{
+				{
+					Long:    "admintoken",
```
Comment (imilchev) [inline providers/atlassian/config/config.go]: to follow the convention set by the other providers this should be `admin-token`

File: providers/atlassian/config/config.go [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+					Desc:    "Provide atlassian hostname (e.g. https://example.atlassian.net).",
+				},
+				{
+					Long:    "user",
+					Type:    plugin.FlagType_String,
+					Default: "",
+					Desc:    "Provide atlassian user name (e.g. example@example.com).",
+				},
+				{
+					Long:    "usertoken",
```
Comment (imilchev) [inline providers/atlassian/config/config.go]: `user-token`

File: providers/atlassian/config/config.go [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+					Type:    plugin.FlagType_String,
+					Default: "",
+					Desc:    "Provide atlassian user api token (used for jira / confluence).",
+				},
+				{
+					Long:    "scimtoken",
```
Comment (imilchev) [inline providers/atlassian/config/config.go]: `scim-token`

File: Makefile (hunk 1 above)
Comment (mariuskimmina) [reply Makefile:318]: It's there

File: providers/atlassian/provider/provider.go [hunk 7]
```diff
@@ -0,0 +1,242 @@
...
+		conf.Options["user"] = string(x.Value)
+	}
+	if x, ok := flags["host"]; ok && len(x.Value) != 0 {
+		conf.Options["host"] = string(x.Value)
+	}
+	if x, ok := flags["admintoken"]; ok && len(x.Value) != 0 {
+		conf.Options["admintoken"] = string(x.Value)
+	}
+	if x, ok := flags["scimtoken"]; ok && len(x.Value) != 0 {
+		conf.Options["scimtoken"] = string(x.Value)
```
Comment (imilchev) [inline providers/atlassian/provider/provider.go]: you can also call the option the same way as the flag, so `scim-token`

File: providers/defaults.go [hunk 8]
```diff
@@ -46,6 +46,18 @@ var DefaultProviders Providers = map[string]*Provider{
 			},
 		},
 	},
+	"atlassian": {
+		Provider: &plugin.Provider{
+			Name:            "atlassian",
+			ConnectionTypes: []string{"atlassian"},
```
Comment (imilchev) [inline providers/defaults.go]: the types should be set here too

**Pull request merged**